pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/MockAuthenticationProvider.java

```diff
@@ -65,4 +71,41 @@ public String authenticate(AuthenticationDataSource authData) throws Authenticat
                 "Not a valid principle. Should be [pass|fail|error].[pass|fail|error], found " + principal);
     }
 
+    @Override
+    public AuthenticationState newAuthState(AuthData authData, SocketAddress remoteAddress, SSLSession sslSession)
+            throws AuthenticationException {
+        return new MockAuthState(this);
+    }
+
+    private static class MockAuthState implements AuthenticationState {
```

Comment:
I think it might make sense to extend this `MockAuthenticationProvider` to get the semantics for the `newAuthState`. I did that with `MockAlwaysExpiredAuthenticationProvider` and with `MockMultiStageAuthenticationProvider`.

My primary concern is that existing tests are likely verifying the behavior of the `OneStageAuthenticationState` class, and changing this method might affect those tests.

Reply:
Make sense.

Reply:
I used `MockAlwaysExpiredAuthenticationProvider` to test authentication refresh because it guaranteed the `ServerCnx` would send the `AuthChallenge` command.